wasm/tests/crypto.test.ts

```diff
@@ -0,0 +1,138 @@
+// external references used for comparison
+import nacl from 'tweetnacl';
+import {sha256} from 'crypto-hash';
+import {performance} from 'perf_hooks';
+import * as crypto from 'crypto';
+
+import {waitReady, ed25519, hasher} from './../';
+
+const toHex = (arrayBuffer: ArrayBuffer | SharedArrayBuffer) => {
+  return Buffer.from(arrayBuffer).toString('hex');
+};
+
+const benchmark = (func: () => void, iterations: number) => {
+  const start = performance.now();
+  for (let i = 0; i < iterations; i++) {
+    func();
+  }
+  const finish = performance.now();
+  return finish - start;
+};
+
+describe('Benchmark in Node.js WASM implementation of ED25519 vs tweetnacl', () => {
+  beforeAll(async () => {
+    await waitReady();
+  });
+
+  // on avg. WASM is 20-25x faster than tweetnacl but I am testing conservative assumption
+  test('WASM sign method should be faster than nacl by at least 15x', () => {
+    const ITERATIONS = 100;
+    const keypair = ed25519.keypair.fromSeed(
+      Uint8Array.from(Array(32).fill(8)),
+    );
+    const data = Uint8Array.from(crypto.randomBytes(1024));
+
+    const wasm = benchmark(() => {
+      ed25519.sign(keypair.publicKey, keypair.secretKey, data);
+    }, ITERATIONS);
+    const mod = benchmark(() => {
+      nacl.sign.detached(data, keypair.secretKey);
+    }, ITERATIONS);
+
+    expect(mod / wasm).toBeGreaterThanOrEqual(15);
+  });
+});
+
+describe('Key generation', () => {
```

Comment:
If we drop the TS wrapper code, we can test the Rust code directly and setup browser integration tests with wasm-bindgen rather than testing here

Reply:
We can add browser integration tests but this test should stay in my opinion as they verify node.js side of the backage.